src/hooks/useBids.ts

```diff
@@ -24,7 +24,20 @@ interface UseBidsReturn {
   error: Error | null
   bids: SaleBid[]
   totalBids: SaleBid[]
-  totalPurchased(bids: any[]): Pick<FixedPriceSalePurchase, 'buyer' | 'amount' | 'status'>[]
+}
+
+export const aggregatePurchases = (bids: SaleBid[], account: string | null | undefined) => {
+  const reduceTotalAmount = bids.reduce((accumulator: any, purchases: any) => {
+    return BigNumber.from(accumulator).add(purchases.amount)
+  }, BigNumber.from(0))
+
+  return [
+    {
+      buyer: account!,
+      amount: reduceTotalAmount,
+      status: bids.length > 0 ? bids[0].status : undefined,
+    },
+  ]
```

Comment:
I am still confused about the purchase of this function. Please document your code from now on.

Why does it return an array when it's aggregating data? This is not ideal.

Reply:
I was taking a shortcut in creating this array. its just easier to simply map over everything rather than stopping and checking to see if its either an array of objects or just an object. 

I will remove the array and update the other components.